src/Functions/FunctionTokens.h

```diff
@@ -129,7 +137,11 @@ class FunctionTokens : public IFunction
                 {
                     size_t token_size = token_end - token_begin;
 
-                    res_strings_chars.resize(res_strings_chars.size() + token_size + 1);
+                    if constexpr (!tokens_maybe_overlapped)
+                        res_strings_chars.resize_assume_reserved(res_strings_chars.size() + token_size + 1);
```

Comment:
In l. 117, we reserve to some size `X`. Here, we resize to `X + token_size + 1` without checking the capacity again, i.e. we will resize beyond the end of the reserved area. That does not sound right.

Reply:
Yes, you are right. Even if tokens_maybe_overlapped is false, we should not assume that the size of `res_strings_chars` doesn't exceed the size of `src_chars`, because extra '\0's will be appended to `res_strings_chars`